### ISSUE TITLE
Hardware - MQTT Client gateway with LAN interface does not support custom clientid / clientid prefix

### ISSUE THREAD

**Issue by whren** [domoticz/domoticz#2935]
MQTT Client gateway with LAN interface does not support custom clientid.
My MQTT broker, mosquitto, is set up with a `clientid_prefixes` filter and so does not allow the domoticz's hardware to connect to it, as the default clientid does not match the filter.

mosquitto.conf:

```configuration
# If set, only clients that have a matching prefix on their
# clientid will be allowed to connect to the broker. By default,
# all clients may connect.
# For example, setting "secure-" here would mean a client "secure-
# client" could connect but another with clientid "mqtt" couldn't.
clientid_prefixes iot-
```

Domoticz setup for the hardware:

![image](https://user-images.githubusercontent.com/1455341/50593519-f1705700-0e98-11e9-919f-be80a6209cf0.png)

Domoticz hardware interface should allow to set a custom clientid or at least a custom clientid prefix.

**Comment by gizmocuz** [comment]
That's correct. You can use an additional username/password and even specify a private CA
Must be 'secure' enough i suppose ? Maybe there is a way in your broker to allow 'your domoticz' client as exception with a username/password and if needed a specific CA ?

**Comment by sweetpants** [comment]
I use this in my /etc/mosquitto/mosquitto.conf

allow_anonymous false
password_file /etc/mosquitto/passwd

/etc/mosquitto/passwd

user:password generated with mosquitto_passwd

Which allows domoticz MQTT gateway to connect to my broker

**Comment by whren** [comment]
Thanks for your feedbacks.
No problem, I'm already aware of the other ways to secure mosquitto but I'm not really looking for a way to disable client id prefixes filtering !

I opened this more as feature request as quite all MQTT clients I use allow either client id customization or prefix setup.

**Comment by StKob** [comment]
So maybe I should bump this issue. When using public mqtt you can't connect to ones which requires clientid. Not always there is option to login without it.

**Comment by gizmocuz** [comment]
Why don't you just add the domoticz mqtt client prefix to the list in the config file?
I do not consider this an issue and since a view versions we have a better client id then we had in the past (random)
Since you are the owner of your broker it is easy
And who uses a public broker to report and control your home?

**Comment by whren** [comment]
@gizmocuz , sorry I've missed your last comment before.

For me it's still an issue as mqtt brokers allows client id specification or prefix filtering in terms of configuration and domoticz not and I'm not sure it's a good thing to be forced to do whitelisting of clients id on the broker side, as it needs supplementary effort to maintain the broker.
Finally I agree with the public broker instance point, but imagine the common use case where you do not own the private broker instance or it is a shared one (a company one for example).
My uses cases are around this and I'm still unable to use this hardware in domoticz because of this.

**Comment by gizmocuz** [comment]
Each client gets a unique id. You can secure the connection with a good password or certificate
